iohk-monitoring/src/Cardano/BM/Data/Severity.lhs

```diff
@@ -54,6 +54,21 @@ data Severity = Debug
               | Emergency
                 deriving (Show, Eq, Ord, Bounded, Enum, Generic, ToJSON, Read)
 
+-- | Severity is a commutative monoid.
+--
+instance Semigroup Severity where
+  Debug     <> _ = Debug
```

Comment:
So `Debug <> Emergency = Debug`?

This is not intuitive to me at all. Say I get a bunch of messages with each a severity, I want them to be traced with the *highest* severity of the bunch, right? Analogy: when you send 3 letters in the same envelope, the priority of the envelope is determined by the letter with the highest priority.

`(<>)` = `max` would make much more sense to me.

Reply:
At the top I argue for `min`:

>  It should compare the message severity with minimum severity found in any configuration settings. This allows to turn on messages from a tracer for better dubugging information, rather than turn them off. That seems like a more sensible default.

Reply:
I see, okay, for combining severities from configs, `min` makes sense (and is confusing, so `(<>)` is indeed better).